councilmatic_core/utils.py

```diff
@@ -0,0 +1,25 @@
+from haystack.utils.highlighting import Highlighter
+
+
+class ExactHighlighter(Highlighter):
+    '''
+    This class customizes the Haystack Highlighter to allow for
+    highlighting multi-word strings.
+    https://django-haystack.readthedocs.io/en/master/highlighting.html#highlighter
+    https://github.com/django-haystack/django-haystack/blob/master/haystack/utils/highlighting.py
+
+    Use this class to build custom filters in `search_result.html`.
+    See LA Metro as a model.
+    '''
+    def __init__(self, query, **kwargs):
+        super(Highlighter, self).__init__()
+        self.query_words = self.make_query_words(query)
+
+    def make_query_words(self, query):
+        query_words = set()
+        if query.startswith('"') and query.endswith('"'):
+            query_words.add(query.strip('\"'))
+        else:
+            query_words = set([word.lower() for word in query.split() if not word.startswith("-")])
+
+        return query_words
```

Comment:
Hmmmm....I am preserving the original behavior of Haystack: https://github.com/datamade/deploy-a-site/blob/master/Daemonizing-applications.md#configure-supervisor-for-your-app

It looks like order persistence does not matter, and the Haystack authors seem to want a set, so that [each word in the collection of query words is unique.](https://github.com/django-haystack/django-haystack/blob/master/haystack/utils/highlighting.py#L37)

Reply:
:+1: Sounds good.